source/_components/switch.rainmachine.markdown

```diff
@@ -13,55 +13,31 @@ ha_iot_class: "Cloud Polling"
 ha_release: 0.51
 ---
 
-The `rainmachine` switch platform allows you to control programs and zones within a [RainMachine smart Wi-Fi sprinkler controller](http://www.rainmachine.com/).
+The `rainmachine` switch platform allows you to control programs and zones
+within a [RainMachine smart Wi-Fi sprinkler controller](http://www.rainmachine.com/). 
 
-## {% linkable_title Configuring the Platform %}
+<p class='note'>
+You must have the [RainMachine component](https://www.home-assistant.io/components/rainmachine/)
+configured to use this switch. After configuring that component, switches will
+automatically appear.
+</p>
 
-The platform allows for either local (i.e., directly across the LAN) or remote (i.e., through RainMachine's cloud API) access; the route you choose will dictate what your configuration should look like.
+## {% linkable_title Configuring the Platform %}
 
-For local access, specify the IP address/hostname of your RainMachine unit, your RainMachine password, and optionally, the device's HTTP port:
+To optionally extend the functionality of these switches, augment your 
+`configuration.yaml` as below:
 
 ```yaml
-switch:
-  platform: rainmachine
+rainmachine:
   ip_address: 192.168.1.100
   password: YOUR_PASSWORD
-```
-
-For remote access, specify your RainMachine username/email and password:
-
-```yaml
-switch:
-  platform: rainmachine
-  email: user@host.com
-  password: YOUR_PASSWORD
+  switches:
+    zone_run_time: 240
 ```
 
 {% configuration %}
-password:
-  description: Your RainMachine password.
-  required: true
-  type: string
-email:
-  description: "Your RainMachine username/email. Cannot be used with the `ip_address` parameter."
-  required: false
-  type: string
-ip_address:
-  description: "The IP address of your RainMachine unit; cannot be used with the `email` parameter."
-  required: optional
-  type: string
-port:
-  description: The TCP port used by your unit for the REST API.
-  required: false
-  type: int
-  default: 8080
-ssl:
-  description: Whether communication with the local device should occur over HTTPS.
-  required: false
-  type: boolean
-  default: true
 zone_run_time:
```

Comment:
Move this to the component page.

Reply:
Not understanding why? This follows a model similar to the Nest docs: https://www.home-assistant.io/components/sensor.nest/ – are you saying that configuration should _only_ be mentioned in the component docs (and not mentioned at all in the platform docs)?

Reply:
Yes, since that's where you change the configuration. It doesn't make sense to mention it on another page.